madmom/evaluation/chords.py

```diff
@@ -0,0 +1,969 @@
+    'maj9': interval_list('(1,3,5,7,9)'),
+    'min9': interval_list('(1,b3,5,b7,9)'),
+    'sus2': interval_list('(1,2,5)'),
+    'sus4': interval_list('(1,4,5)'),
+    '11': interval_list('(1,3,5,b7,9,11)'),
+    'min11': interval_list('(1,b3,5,b7,9,11)'),
+    '13': interval_list('(1,3,5,b7,13)'),
+    'maj13': interval_list('(1,3,5,7,13)'),
+    'min13': interval_list('(1,b3,5,b7,13)')
+}
+
+
+def chord_intervals(quality_str):
+    """
+    Convert a chord quality string to a pitch class representation. For
+    example, 'maj' becomes [1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0].
+
+    Parameters
+    ----------
+    quality_str : str
+        String defining the chord quality.
+
+    Returns
+    -------
+    pitch_classes : numpy array
+        Binary pitch class representation of chord quality.
+
+    """
+    list_idx = quality_str.find('(')
+    if list_idx == -1:
+        return _shorthands[quality_str].copy()
+    if list_idx != 0:
+        ivs = _shorthands[quality_str[:list_idx]].copy()
+    else:
+        ivs = np.zeros(12, dtype=np.int)
+
+    return interval_list(quality_str[list_idx:], ivs)
+
+
+def load_chords(filename):
+    """
+    Load chords from a text file.
+
+    The chord must follow the syntax defined in [1]_.
+
+    Parameters
+    ----------
+    filename : str
+        File containing chord segments.
+
+    Returns
+    -------
+    crds : numpy structured array
+        Structured array with columns "start", "end", and "chord",
+        containing the beginning, end, and chord definition of chord
+        segments.
+
+    References
+    ----------
+    .. [1] Christopher Harte, "Towards Automatic Extraction of Harmony
+           Information from Music Signals." Dissertation,
+           Department for Electronic Engineering, Queen Mary University of
+           London, 2010.
+
+    """
+    # TODO: Join consecutive labels of identical chords
+    start, end, chord_labels = [], [], []
+    with open(filename, 'r') as f:
+        for line in f:
+            s, e, l = line.split()
+            start.append(float(s))
+            end.append(float(e))
+            chord_labels.append(l)
+
+    crds = np.zeros(len(start), dtype=CHORD_ANN_DTYPE)
+    crds['start'] = start
+    crds['end'] = end
+    crds['chord'] = chords(chord_labels)
+    return crds
+
+
+def evaluation_pairs(det_chords, ann_chords):
+    """
+    Match detected with annotated chords and create paired label segments
+    for evaluation.
+
+    Parameters
+    ----------
+    det_chords : numpy structured array
+        Chord detections with 'start' and 'end' fields.
+
+    ann_chords : numpy structured array
+        Chord annotations with 'start' and 'end' fields.
+
+    Returns
+    -------
+    annotations : numpy structured array
+        Annotated chords of evaluation segments.
+    detections : numpy structured array
+        Detected chords of evaluation segments.
+    durations : numpy array
+        Durations of evaluation segments.
+    """
```

Comment:
Minor thing: I usually add an empty line before the closing `"""` -- as you did in many other places.
I just mention it here, it misses multiple times.